src/commands/force/org/display.ts

```diff
@@ -61,14 +61,15 @@ export class OrgDisplayCommand extends SfdxCommand {
       sfdxAuthUrl: this.flags.verbose && fields.refreshToken ? authInfo.getSfdxAuthUrl() : undefined,
       alias: await getAliasByUsername(fields.username),
     };
+    this.ux.warn(sharedMessages.getMessage('SecurityWarning'));
+
     if (!this.flags.json) {
       this.print(returnValue);
     }
     return returnValue;
   }
 
   private print(result: OrgDisplayReturn): void {
-    this.ux.warn(sharedMessages.getMessage('SecurityWarning'));
```

Comment:
@mshanemc Not issuing the message when `--json` output requested was intentional.